src/project-manager.ts

```diff
@@ -472,39 +482,67 @@ export class ProjectManager implements Disposable {
 	 * If there is no root configuration, adds it to catch all orphan files
 	 */
 	createConfigurations() {
-		const rootdirs = new Set<string>();
 		for (const uri of this.localFs.uris()) {
-			const relativeFilePath = path_.posix.relative(this.rootPath, util.uri2path(uri));
-			if (!/(^|\/)[tj]sconfig\.json$/.test(relativeFilePath)) {
+			const filePath = util.uri2path(uri);
+			if (!/(^|\/)[tj]sconfig\.json$/.test(filePath)) {
 				continue;
 			}
-			if (/(^|\/)node_modules\//.test(relativeFilePath)) {
+			if (/(^|\/)node_modules\//.test(filePath)) {
 				continue;
 			}
-			let dir = path_.posix.dirname(relativeFilePath);
-			if (dir === '.') {
+			let dir = util.toUnixPath(filePath);
+			const pos = dir.lastIndexOf('/');
+			if (pos <= 0) {
 				dir = '';
+			} else {
+				dir = dir.substring(0, pos);
+			}
+			const configType = this.getConfigurationType(filePath);
+			let configs = this.configs.get(configType);
+			if (!configs) {
+				configs = new Map<string, ProjectConfiguration>();
+				this.configs.set(configType, configs);
```

Comment:
We only have two fixed types of configurations. We should initialize these maps in the constructor. It might be even easier (and faster) to have simply two attributes, `jsConfigs` and `tsConfigs`.